### ISSUE TITLE
Animal Crossing: New Leaf - Objects in distance disappear / become black

### ISSUE THREAD

**Issue by Leo626** [citra-emu/citra#1514]
As the player moves away from the mountain in the distance it becomes black. Also notice the fence becomes black as well.
Using master as of afb5140

![2660eeee-e853-11e5-99b4-59a6d7584924](https://cloud.githubusercontent.com/assets/17358613/13732436/250b3ae8-e959-11e5-9b8e-f2f0750e7bf7.gif)

[LOG](https://gist.github.com/Leo626/8e90d3dcb3d7c0e86038)

**Comment by Zaneo** [comment]
Aside from the fact that it was black, as opposed to sky coloured, is this not the correct behaviour?

![ezgif com-gif-maker](https://cloud.githubusercontent.com/assets/295949/13733471/b528fb5a-e969-11e5-82ed-e5cebb0ae7ba.gif)

**Comment by yuriks** [comment]
It's hard to tell, but it seems that in your gif the background simply pans out of view, while in the Citra one it clips out. This might be caused by us not implementing depth scale. (cc @tfarley)

**Comment by Punk34** [comment]
This happens in many games. I think AC uses some sort of fog, which is not rendered on citra.

**Comment by JayFoxRox** [comment]
Test with #1621 please

**Comment by Leo626** [comment]
@JayFoxRox Tested the w-buffer branch and it remains unchanged. However, I'm not sure if I built it wrong but mines had a memory leak. Citra-qt was consuming 6.9GB out 8GB of RAM. Can someone else try it?

**Comment by KillzXGaming** [comment]
Mine uses 4-5gb out of 8gb RAM. Animal Crossing gets big framerate drop, as neared full speed before, however sky is fixed. No idea if particles worked before but now they do, footsteps sorta glitch but show up. Will post screenshots.

**Comment by Leo626** [comment]
Tried the new w-buffer commits with master fc6f898 and nothing's changed. I don't know if I'm building it wrong but I still get memory leaks. 

On a side-note, turning off Hardware Renderer fixed the depth scaling and black arm issue (#1515).

![citra-qt_2016-03-31_21-35-06](https://cloud.githubusercontent.com/assets/17358613/14195608/2aca15a6-f789-11e5-867a-79401c2cb102.png)

**Comment by JayFoxRox** [comment]
Also try #1624 .
@Leo626 post all responses in the respective issues. It's hard to follow issues otherwise.
If there is speculation it could also end up on a fast paced medium such as IRC (\#citra-dev) or once it is resolved everyone involved could delete their posts if they don't contribute anything meaningful. kthxbye

**Comment by Leo626** [comment]
Hi @JayFoxRox, tried it. The depth scaling issue is still there but the memory leak has been fixed.

**Comment by Leo626** [comment]
Fixed with master aab4160 merged with #1621  and #1690.

![citra-qt_2016-04-30_18-30-40](https://cloud.githubusercontent.com/assets/17358613/14938933/9bff7604-0f02-11e6-943d-59b082d6a394.png)

**Comment by Leo626** [comment]
I deleted my previous post because I jumped to conclusions to quickly. This issue is fixed with aab4160.

![citra-qt_2016-04-30_19-06-37](https://cloud.githubusercontent.com/assets/17358613/14939100/e6c470d2-0f06-11e6-8fd9-55773079f94e.png)
